src/worker/event/registry.go

```diff
@@ -0,0 +1,157 @@
+package event
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-lambda/open-lambda/ol/common"
+)
+
+// UploadLambda handles POST /registry/{name} - saves to common.Conf.Registry
+func UploadLambda(w http.ResponseWriter, r *http.Request) {
+	funcName := strings.TrimPrefix(r.URL.Path, REGISTRY_BASE_PATH)
```

Comment:
should there be any locking at this level, to prevent concurrent upload of the same file?